src/mount.rs

```diff
@@ -192,11 +192,8 @@ impl<'a> MountOptions<'a> {
         let image = std::fs::File::open(self.image)?;
 
         if let Some(expected) = self.digest {
-            let measured: fsverity::Sha256HashValue =
-                fsverity::ioctl::fs_ioc_measure_verity(&image)?;
-            if expected != hex::encode(measured) {
-                panic!("expected {:?} measured {:?}", expected, measured);
-            }
+            let expected = sha256_from_str(expected)?;
+            fsverity::compare_verity(&image, &expected)?;
```

Comment:
Looking at it at a call site I think the "compare" verb is wrong.  I'd expect that for a function with a boolean return value.  We probably want something along the lines of "ensure" "check" "assert" etc. that implies that it's an error if it doesn't match.